File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:

        res, maxCount, maxLength = 0, -inf, -inf
        arr = [0]*26

        left = 0
        for right in range(len(s)):
            arr[ord(s[right])-ord('A')] += 1

            maxCount = max(maxCount, arr[ord(s[right])-ord('A')])

            while right-left-maxCount+1 > k:
                arr[ord(s[left])-ord('A')] -= 1
                left += 1        

            print(right)
            print(left)

            maxLength = max(maxLength, right-left+1)
            
        return maxLength

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {

        int maxLength = 0;
        int maxCount = 0;
        int[] arr = new int[26];


        int left = 0;
        for(int right = 0; right < s.length(); right++){
            arr[s.charAt(right) - 'A']++;

            maxCount = Math.max(maxCount, arr[s.charAt(right) - 'A']);

            while(right-left+1-maxCount > k){
                arr[s.charAt(left) - 'A']--;                
                left++;
            }
            maxLength = Math.max(maxLength, right-left+1);
        }

        return maxLength;        
    }
}